pkg/lite-apiserver/cache/cache_mgr.go

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright 2020 The SuperEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/runtime/serializer/streaming"
+	"k8s.io/apimachinery/pkg/watch"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclientwatch "k8s.io/client-go/rest/watch"
+	"k8s.io/klog"
+
+	"github.com/superedge/superedge/pkg/lite-apiserver/constant"
+	"github.com/superedge/superedge/pkg/lite-apiserver/storage"
+)
+
+type CacheManager struct {
+	storage storage.Storage
+}
+
+func NewCacheManager(storage storage.Storage) *CacheManager {
+	return &CacheManager{
+		storage: storage,
+	}
+}
+
+func (c CacheManager) Cache(req *http.Request, statusCode int, header http.Header, body io.ReadCloser) error {
+	info, ok := apirequest.RequestInfoFrom(req.Context())
+	if !ok {
+		return fmt.Errorf("parse requestInfo error")
+	}
+
+	userAgent := getUserAgent(req)
+	key := cacheKey(userAgent, info)
+	klog.V(4).Infof("cache for %s", key)
+
+	return c.handleCache(key, info.Verb, statusCode, header, body)
```

Comment:
Could ignore all non-200 responses.

Reply:
we only cache Http.OK now, filter this in proxy.modifyResponse()